src/arkode/arkode_impl.h

```diff
@@ -287,6 +288,11 @@ typedef int (*ARKTimestepAttachMasssolFn)(
 typedef void (*ARKTimestepDisableMSetup)(ARKodeMem ark_mem);
 typedef void* (*ARKTimestepGetMassMemFn)(ARKodeMem ark_mem);
 
+/* time stepper interface functions -- forcing */
+typedef int (*ARKTimestepSetForcingFn)(void* arkode_mem, sunrealtype tshift,
```

Comment:
All the other typedefs use `ARKodeMem` as the first argument. Also eliminates some casting in the implementation